api/proto/teleport/legacy/types/types.proto

```diff
@@ -5994,6 +5998,28 @@ message PluginEntraIDSyncSettings {
   repeated string default_owners = 1;
 }
 
+// AccessGraphSettings controls settings for syncing access graph specific data.
+message PluginEntraIDAccessGraphSettings {
+  option (gogoproto.equal) = true;
+
+  // AppSsoSettingsCache is a map where keys are Entra Enterprise Application "app IDs",
+  // and values are single sign-on settings for the enterprise application indentified by that ID.
+  //
+  // This data is stored here because it is not available through traditional methods (MS Graph API).
+  // Instead, it is fetched once during the plugin's set up using the user's credentials to connect to Azure's private API.
+  map<string, PluginEntraIDAppSSOSettings> app_sso_settings_cache = 1;
```

Comment:
can we change this to repeated field? 
IF we need to ensure uniqueness, we can do it manually

Reply:
We can, and then have a map at runtime instead. Are there known pitfalls with map fields in proto? I see that we have quite a few, but mostly for labels.

Reply:
@tigrato done in c59a322